### DISCUSSION TITLE
Tag protection w/ Actions: `github.ref_protected` is `false` for tag matching protection rule

### DISCUSSION THREAD

**Discussion by codingchipmunk** [community/community#66588] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi,
I have configured a GH action to build and release a Python lib. The action is triggered on pushes of tags matching `v*`. I have also setup tag protections using the rule `v*`.
As an additional safeguard against misconfigurations or other funky stuff, I added a step to the action that will only execute if the condition `github.ref_protected` is `false` and always fails (meaning the action will only execute if the trigger was protected).

When I push a tag like `v0.1.0`, the flow is triggered, but the safeguard is tripped because `github.ref_protected`  is evaluating to `false`. So either the tag is not protected, even though it should be based on my understanding, or something else is going on.

I am aware of #62991, but I don't think it is related.

I would appreciate it if someone could help me with figuring out what I am doing wrong here.

**Comment by rcwbr** [comment]
I am seeing the same behavior and have narrowed my case far enough to have confidence that this is a bug. See https://github.com/orgs/community/discussions/142985
